utilities/utils/utils.go

```diff
@@ -51,6 +51,9 @@ func IsNoOp(data model.ProcessData) bool {
 }
 
 func AddLabel(config *container.Config, key string, value string) {
+	if config.Labels == nil {
```

Comment:
I already have initialized map before we setup any fields in here .https://github.com/rancher/agent/blob/f4f3c0cb0d6518db3336d4f9d15c687d0bef5f04/core/compute/compute_common.go#L78